JetUtilities/src/L2Creator.cc

```diff
@@ -431,372 +550,374 @@ bool L2Creator::getL3Rsp() {
 void L2Creator::writeTextFileForCurrentAlgorithm() {
-	TString txtfilename = outputDir + era + "_L2Relative_" + ji->getAlias() + ".txt";
-	ofstream fout(txtfilename);
-	fout.setf(ios::right); 
-
-	unsigned int vector_size = 0;
-    if(l2l3) vector_size = vabscor_eta.size(); //For L2L3 Corrections Together
-    else vector_size = vrelcor_eta.size(); //For L2 & L3 Corrections Separate
-    for (unsigned int ieta=0;ieta<vector_size;ieta++) { 
-    	TGraph* grelcor;
-      	if(l2l3) grelcor = vabscor_eta[ieta]; //For L2L3 Corrections Together
-      	else grelcor = vrelcor_eta[ieta]; //For L2 & L3 Corrections Separate
-      	TF1* frelcor = (TF1*)grelcor->GetListOfFunctions()->Last();
-      	if(frelcor!=0) {
-      		if(ieta==0 || (ieta==1 && delphes))
-               fout<<"{1 JetEta 1 JetPt max(0.0001,"<<frelcor->GetTitle()<<") Correction L2Relative}"<<endl;
-               //fout<<"{1 JetEta 1 JetPt max(0.0001,"<<frelcor->GetExpFormula()<<") Correction L2Relative}"<<endl;
-      		double  etamin  = hl_jetpt.minimum(0,ieta);
-      		double  etamax  = hl_jetpt.maximum(0,ieta);
-      		double  ptmin = grelcor->GetX()[0];
-      		double  ptmax = grelcor->GetX()[grelcor->GetN()-1];
-      		fout<<setw(11)<<etamin
-      		<<setw(11)<<etamax
-        	    <<setw(11)<<(int)(frelcor->GetNpar()+2) //Number of parameters + 2 
-        	    <<setw(12)<<ptmin
-        	    <<setw(12)<<ptmax;
-        	    for(int p=0; p<frelcor->GetNpar(); p++) {
-        	    	fout<<setw(13)<<frelcor->GetParameter(p);
-        	    }
-        	    fout<<endl;
-        }
+  TString txtfilename = outputDir + "/" + era + "_L3Absolute_" + ji->getAlias() + ".txt";
+  ofstream fout(txtfilename);
+  fout.setf(ios::right); 
+
+  unsigned int vector_size = 0;
+  if(l2l3) vector_size = vabscor_eta.size(); //For L2L3 Corrections Together
+  else vector_size = vrelcor_eta.size(); //For L2 & L3 Corrections Separate
+  for (unsigned int ieta=0;ieta<vector_size;ieta++) { 
+    TGraph* grelcor;
+    if(l2l3) grelcor = vabscor_eta[ieta]; //For L2L3 Corrections Together
+    else grelcor = vrelcor_eta[ieta]; //For L2 & L3 Corrections Separate
+    TF1* frelcor = (TF1*)grelcor->GetListOfFunctions()->Last();
+    if(frelcor!=0) {
+      if(ieta==0 || (ieta==1 && delphes)){
+        if(cbPar==17) fout<<"{1 JetEta 1 JetPt max(-1.,"<<frelcor->GetExpFormula()<<") Correction L3Absolute}"<<endl;
+	else          fout<<"{1 JetEta 1 JetPt max(0.0001,"<<frelcor->GetExpFormula()<<") Correction L3Absolute}"<<endl;
+	}
+      double  etamin  = hl_jetpt.minimum(0,ieta);
+      double  etamax  = hl_jetpt.maximum(0,ieta);
+      double  ptmin = grelcor->GetX()[0];
+      double  ptmax = grelcor->GetX()[grelcor->GetN()-1];
+      fout<<setw(8)<<etamin
+	<<setw(8)<<etamax
+	<<setw(6)<<(int)(frelcor->GetNpar()+2) //Number of parameters + 2 
+	<<setw(12)<<ptmin
+	<<setw(12)<<ptmax;
+      for(int p=0; p<frelcor->GetNpar(); p++) {
+	fout<<setw(16)<<frelcor->GetParameter(p); 
```

Comment:
Yes I have more significant digits. My Eta bins are [1] so adding two more digit is ok.:

[1] 
etaBinning = [ -2.3, -2.1, -1.9, -1.7, -1.5, -1.3, -1.1, -0.9, -0.7, -0.5, -0.3, -0.1,
                +0.1, +0.3, +0.5, +0.7, +0.9, +1.1, +1.3, +1.5, +1.7, +1.9, +2.1, +2.3 ]